### PR TITLE
feat: integrate formbricks in help feedback box

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
The Help Feedback Box currently sends the data to our email service. This PR sends it to a pre-created Formbricks Survey for easier visibility and availability in the Formbricks App.

image

We use the `@formbricks/api` package for minimal dependencies and create a response in the app.

## Requirement/Documentation
We introduce 2 new environment variables that needs to be set:
1. `FORMBRICKS` in the form of `[apiHost]+[environmentId]`
2. `FORMBRICKS_FEEDBACK_SURVEY_ID` in the form of `surveyId`

For testing, use:
```
FORMBRICKS=https://app.formbricks.com+cloob69in56synz0fg6l5vqmx
FORMBRICKS_FEEDBACK_SURVEY_ID=cloob6fn356u0nz0fl0mm1wa5
```

Before merging to main, 
- [x] Create a Survey on Formbricks
- [x] Create 2 questions (1 Open Text with Question ID as: `formbricks-share-comments-question` and 1 Rating question with Range of 4  and ID as `formbricks-rating-question` )

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- Set the env variable as suggested above
- Fill the Feedback Box inside the Help Button
- View in the Formbricks App and the response should be reflected

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by PeerRich — APPROVED** [review]
looks very good and minimal, will try today!

**Comment by PeerRich** [comment]
lets also add userid and email

**Comment by PeerRich** [comment]
added .env to both prod and staging, added formbricks survey with the question ID

File: .env.example
Comment (PeerRich) [inline .env.example:124]: does this have to be FORMBRICKS or NEXT_PUBLIC_FORMBRICKS ?

also, doesn't this need to be added to turbo.json?

**Review by alishaz-polymath — DISMISSED** [review]
Tested, works!
NIT: We could also add userId and/or email as a hidden field (auto-populates internally).

Thank you for your contribution 🚀

**Comment by ShubhamPalriwala** [comment]
Hey, thanks for the quick reviews and testing! Glad it works for everyone 🚀 
We're currently updating our API endpoint to support the linking of userId and email as well via the API. Will ping the reviewers here once the PR is again ready for review

**Comment by PeerRich** [comment]
draft until API is ready for username and email

**Review by PeerRich — DISMISSED** [review]
env and keys added to vercel prod

File: .env.example
Comment (zomars) [inline .env.example:125]: Not quite since is used on the server only. But I would go with something more descriptive.

```suggestion
FORMBRICKS_URL=
```

Comment (zomars) [inline .env.example:125]:
```suggestion
# Formbricks Experience Management Integration
FORMBRICKS_HOST_URL=
FORMBRICKS_ENVIRONMENT_ID=
```

File: packages/lib/formbricks.ts
Comment (zomars) [inline packages/lib/formbricks.ts:11]:
```suggestion
if (!process.env. FORMBRICKS_HOST_URL || !process.env. FORMBRICKS_ENVIRONMENT_ID) throw new Error("Missing FORMBRICKS variables");
const apiHost = process.env.FORMBRICKS_HOST_URL;
const environmentId = process.env.FORMBRICKS_ENVIRONMENT_ID;
```

File: packages/trpc/server/routers/loggedInViewer/submitFeedback.handler.ts
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/submitFeedback.handler.ts:34]:
```suggestion
  if (process.env. FORMBRICKS_HOST_URL && process.env. FORMBRICKS_ENVIRONMENT_ID) sendFeedbackFormbricks(rating, comment);
```

File: turbo.json
Comment (zomars) [inline turbo.json:241]:
```suggestion
    "FORMBRICKS_ENVIRONMENT_ID",
    "FORMBRICKS_FEEDBACK_SURVEY_ID",
    "FORMBRICKS_HOST_URL",
```

**Review by zomars — DISMISSED** [review]
Thank you for your contribution 🙏 

Quick note. Let's prioritize legibility over clever code in this case. Left some suggestions to address.

**Comment by ShubhamPalriwala** [comment]
Hey folks, the PR is once again ready for review 🚀 We now add the username of the person as a `userId` as a person and attach it to the response submitted! We do not support the addition of custom attributes in the same response creation API yet hence going ahead with just the username for now. Please take a look 🙏🏼 

Also addressed the env var changes as suggested

File: .env.example
Comment (PeerRich) [inline .env.example:124]:
```suggestion
FORMBRICKS_HOST_URL=https://app.formbricks.com
```

**Comment by ShubhamPalriwala** [comment]
update, discussed with @mattinannt and restructuring things a bit

**Comment by mattinannt** [comment]
@PeerRich During our testing, we discovered a potential security issue with the Formbricks API updates we pushed to enable this PR. We will update you as soon as this is fixed. You can review this before then, but please don't merge yet.

**Review by PeerRich — DISMISSED** [review]

**Review by zomars — APPROVED** [review]
LGTM

**Comment by zomars** [comment]
We need to fix types tho 
image

**Review by zomars — DISMISSED** [review]
Type checks are failing

**Comment by ShubhamPalriwala** [comment]
Thanks for the feedback @zomars, I've made the type fixes (apologies for overlooking it) and tested the flow again 💪🏼 
cc: @PeerRich

**Review by PeerRich — APPROVED** [review]

**Pull request merged**
